### PR TITLE
perf(ext/ffi): optimize synchronous calls 

### PR DESCRIPTION
**Pull request by littledivy**
This commit implements an optimization
for blocking FFI calls. 

`op_ffi_load` creates JS functions for 
synchronous symbols. This avoids 
reallocations that existed in `op_ffi_call`
and allows future V8 fast API optimizations.

Each function owns an allocation to corresponding
the Symbol, which is freed on GC.

Nonblocking ops & statics are still registered JS
side and use ops to make calls.

image

### CODE REVIEW

**Review by bartlomieju — APPROVED** [review]
LGTM, these are very impressive results

**Comment by lucacasonato** [comment]
I want to discuss this in CLI design meeting, because it bypasses ops.

**Comment by lucacasonato** [comment]
Also unintentional SWC downgrade?

File: ext/ffi/lib.rs [hunk 1]
```diff
@@ -541,24 +501,111 @@ where
...
+  let func = v8::Function::builder(
+    |scope: &mut v8::HandleScope,
+     args: v8::FunctionCallbackArguments,
+     mut rv: v8::ReturnValue| {
+      let external: v8::Local<v8::External> =
+        args.data().unwrap().try_into().unwrap();
+      // SAFETY: The pointer will not be deallocated until the function is
+      // garbage collected.
+      let symbol = unsafe { &*(external.value() as *const Symbol) };
+      let result = ffi_call_sync(scope, args, symbol).unwrap();
```
Comment (aapoalas) [inline ext/ffi/lib.rs:588]: Should handle errors.

**Comment by bartlomieju** [comment]
> Also unintentional SWC downgrade?

How so?

File: ext/ffi/lib.rs (hunk 1 above)
Comment (bartlomieju) [reply ext/ffi/lib.rs:588]: Good point, I've missed that

**Review by bartlomieju — CHANGES_REQUESTED** [review]
https://github.com/denoland/deno/pull/14945#discussion_r904866038

**Comment by lucacasonato** [comment]
> How so?

Look at cli/Cargo.toml

**Comment by bartlomieju** [comment]
> > How so?
> 
> Look at cli/Cargo.toml

Screenshot 2022-06-23 at 12 42 49

I see only two changed files

**Comment by lucacasonato** [comment]
Huh, on the GitHub mobile app I see a bunch more changed files. I guess must be a bug. It looks correct on the website... sorry for the noise.

**Review by bartlomieju — APPROVED** [review]

**Pull request merged**